.github/workflows/tiiuae-pixhawk.yaml

```diff
@@ -132,3 +132,28 @@ jobs:
           jfrog rt bpr "$BUILD_NAME_DEB" "$GITHUB_SHA" "$ARTIFACTORY_DEB_REPO" \
                        --status dev \
                        --comment "development build"
+
+      - name: Firmware flasher - Container metadata
+        id: containermeta # referenced from later steps
+        uses: docker/metadata-action@v3
+        with:
+          images: ghcr.io/tiiuae/px4-firmware
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=sha
+
+      - name: Firmware flasher - Prepare for build
+        # If we'd run build in Tools/ directory, I don't know how to pass both bin/ and Tools/ as context
+        run: |
+          cp px4-firmware/Tools/px_uploader.Dockerfile Dockerfile
```

Comment:
This is a bit of a gross trick, so let me know if you have better ideas

Reply:
You can use "file:" key to use dockerfile with custom name

```
with:
          push: true
          context: .
          file: px4-firmware/Tools/px_uploader.Dockerfile
          tags: ${{ steps.containermeta.outputs.tags }}
          labels: ${{ steps.containermeta.outputs.labels }}
```

Reply:
Nice, thanks!

I knew you could give the file, but assumed build context was relative to that, but apparently the `context: .` keeps the context at repo root, so everything works out!

Thanks again!